src/optimization/CasADiOpti.cpp

```diff
@@ -28,8 +28,12 @@ void CasADiOpti::SetInitial(const casadi::MX& expression, double value) {
 [[nodiscard]]
 expected<void, std::string> CasADiOpti::Solve(bool diagnostics) {
   GetCancellationFlag() = 0;
-  const auto callback =
-      new const CasADiIterCallback("f", opti.nx(), opti.ng(), opti.np());
+  const auto callback = new const CasADiIterCallback(
+      "f", opti.nx(), opti.ng(), opti.np(), [=, this]() {
+        for (auto it = callbacks.begin(); it < callbacks.end(); it++) {
+          (*it)();
```

Comment:
```suggestion
        for (auto& c : callbacks) {
          c();
```
Using c cuz callback is already taken in the outer scope, which would be confusing.